subsys/bootloader/provision/provision_flash.c

```diff
@@ -45,9 +50,21 @@ int public_key_data_read(u32_t key_idx, u8_t *p_buf, size_t buf_size)
 		return -EINVAL;
 	}
 
-	for (size_t i = 0; i < CONFIG_SB_PUBLIC_KEY_HASH_LEN; i++) {
-		p_buf[i] = p_provision_data->pkd[(key_idx *
-				CONFIG_SB_PUBLIC_KEY_HASH_LEN) + i];
+	p_key = &(p_provision_data->pkd[key_idx *
+			CONFIG_SB_PUBLIC_KEY_HASH_LEN]);
+
+#ifdef CONFIG_SOC_NRF9160
+	/* Ensure word alignment, as provision data is stored in memory region
+	 * with word sized read limitation. Perform both build time and run
+	 * time asserts to catch the issue as soon as possible.
+	 */
+	BUILD_ASSERT(CONFIG_SB_PUBLIC_KEY_HASH_LEN % 4 == 0);
+	BUILD_ASSERT(offsetof(provision_flash_t, pkd) % 4 == 0);
+	__ASSERT((p_key % 4 == 0), "Key address is word aligned");
```

Comment:
The message is for failures, right?

"Key address is not word aligned"

Reply:
Correct